repair/repair.cc

```diff
@@ -1480,3 +1493,52 @@ future<> repair_abort_all(seastar::sharded<database>& db) {
         repair_tracker().abort_all_repairs();
     });
 }
+
+future<> sync_data_with_repair(seastar::sharded<database>& db,
+        sstring keyspace,
+        dht::token_range_vector ranges,
+        std::unordered_map<dht::token_range, repair_neighbors> neighbors) {
+    if (ranges.empty()) {
+        return make_ready_future<>();
+    }
+    return smp::submit_to(0, [&db, keyspace = std::move(keyspace), ranges = std::move(ranges), neighbors = std::move(neighbors)] () mutable {
```

Comment:
In the past, always submitting to CPU0 had led to issues.
During a bootstrap operation, for instance, CPU0 will become overloaded and lead to problems like the one documented  here : https://github.com/scylladb/scylla/issues/3156

We have a vector of ranges, which are hashes. Please use one such range to consistently select a random CPU to coordinate the process.

Reply:
1. The work submitted to shard zero is lightweight. It only creates repair metadata and asks all the shards to run.
2. This work is once per keyspace during node operations. We call it one after another.
3. We summit it to shard zero because the repair_tracker which controls all the repair instances works on shard zero only. It is per node object.

Reply:
Does it become heavier if it is sent to other shards?

If not, do it.
Every thing blindly sent to shard0 is, on its own, lightweight. But together they add up.

Reply:
See my item 3 above. We have per node object that works only on shard zero only. If we submit to a random shard, we need to bounce back to shard zero to access it, which makes the whole operation heavier. 

Yes, they add up, but why does it matter if it is one time (and lightweight) operation? 

If we want to get rid of the node object, aka, repair_tracker, it is beyond this series.

Reply:
Agree it's beyong the scope of the series.

How many calls to shard 0 do we have per repair?